src/MatrixTypes_PETSc.f90

```diff
@@ -565,6 +568,20 @@ SUBROUTINE matmult_PETSc(A,B,C,alpha,beta,transA,transB)
       DEALLOCATE(tmpB)
       DEALLOCATE(tmpC)
     ENDSUBROUTINE matmult_PETSc
+!
+!-------------------------------------------------------------------------------
+!> @brief set all the entries of the matrix to be zero
+!> @param matrix declare the matrix type to act on
+!>
+!>
+    SUBROUTINE zeroentries_PETScMatrixType(matrix)
+      CHARACTER(LEN=*),PARAMETER :: myName='zeroentries_PETScMatrixType'
+      CLASS(PETScMatrixType),INTENT(INOUT) :: matrix
+      PetscErrorCode  :: ierr
+      IF(matrix%isInit) THEN
```

Comment:
Change to a DBC REQUIRE

Are there any other requirements on the state of the matrix for this? E.g. prealloced, assembled, etc. If so, those need to be checked.

Reply:
I believe as long as the matrix type is preallocation. Since the initiation of PETSC type finished everything, so there's no need to check preallocation.